src/api/ITooltip.js

```diff
@@ -60,6 +61,13 @@
         this.tooltip = d3Tip()
             .attr("class", "d3-tip")
             .offset(function (d) {
+                if(event && context.tooltipFollowMouse()){
+                    var d3tipElement = document.querySelector(".d3-tip"); // d3Tip offers no reference to the '.d3-tip' element...?
+                    d3tipElement.style.display = "block";
+                    d3tipElement.style.left = context.tooltipOffset() + (event.clientX) + "px";
+                    d3tipElement.style.top = event.clientY + "px";
+                    return [];
```

Comment:
Why set the element top + left instead of returning the offset in the array?

Reply:
Should you take the "direction" into consideration?

Reply:
The offset is based on `d3-tip`'s determined placement. Setting the `left` and `top` allow us to position it based on its relative parent element.

Reply:
In order to take direction into consideration we would need to getBoundingClientRect on the resulting tooltip html. The placement happens on mouseover tick so this would be very expensive... that said... we could cache the resulting rect based on a hash of the tooltip html and only have to call getBoundingClientRect once for each series. Thoughts?